tests/model_registry/test_model_registry_scc.py

```diff
@@ -0,0 +1,149 @@
+            client=admin_client, pod_prefix=MR_INSTANCE_NAME, namespace=model_registry_namespace.name
+        )
+        if len(pods) != 1:
+            pytest.fail(
+                "Expected one model registry pod. Found: {[{pod.name: pod.status} for pod in pods] if pods else None}"
+            )
+        return pods[0]
+    else:
+        raise AssertionError(f"Invalid resouce requests: {request.param['kind']}")
+
+
+@pytest.mark.parametrize(
+    "model_registry_namespace, updated_dsc_component_state_scope_class, registered_model",
+    [
+        pytest.param(
+            {"namespace_name": MR_NAMESPACE},
+            {
+                "component_patch": {
+                    DscComponents.MODELREGISTRY: {
+                        "managementState": DscComponents.ManagementState.MANAGED,
+                        "registriesNamespace": MR_NAMESPACE,
+                    },
+                },
+            },
+            MODEL_DICT,
+        ),
+    ],
+    indirect=True,
+)
+@pytest.mark.usefixtures("model_registry_namespace", "updated_dsc_component_state_scope_class", "registered_model")
+class TestModelRegistrySecurityContextValidation:
+    @pytest.mark.parametrize(
+        "model_registry_resource",
+        [
+            pytest.param({"kind": Deployment}),
+        ],
+        indirect=["model_registry_resource"],
+    )
+    def test_model_registry_deployment_security_context_validation(
+        self: Self,
+        model_registry_resource: NamespacedResource,
+        model_registry_namespace,
+    ):
+        """
+        Validate that model registry deployment does not set runAsUser/runAsGroup
+        """
+        error = []
+        for container in model_registry_resource.instance.spec.template.spec.containers:
+            if not all([True for key in KEYS_TO_VALIDATE if not container.get(key)]):
+                error.append({container.name: container.securityContext})
+
+        if error:
+            pytest.fail(
+                f"{model_registry_resource.name} {model_registry_resource.kind} containers expected to not "
+                f"set {KEYS_TO_VALIDATE}, actual: {error}"
+            )
+
+    @pytest.mark.parametrize(
+        "model_registry_resource",
+        [
+            pytest.param({"kind": Pod}),
+        ],
+        indirect=["model_registry_resource"],
+    )
+    def test_model_registry_pod_security_context_validation(
+        self: Self,
+        model_registry_resource: NamespacedResource,
+        model_registry_scc_namespace: dict[str, str],
+    ):
+        """
+        Validate that model registry pod gets runAsUser/runAsGroup from openshift and the values matches namespace
+        annotations
+        """
+        errors = []
+        uid_ranges = model_registry_scc_namespace["uid-range"].split("/")
+        ns_uid = uid_ranges[0]
+        pod_spec = model_registry_resource.instance.spec
+        pod_selinux_option = pod_spec.securityContext.get("seLinuxOptions", {}).get("level")
```

Comment:
please split to functions by tested functionality; it is hard to follow what is being tested